src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@Service
+public class NetworkModificationApplicator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkModificationApplicator.class);
+
+    private enum ApplicationMode {
+        UNITARY,
+        MULTIPLE
+    }
+
+    private static final String NETWORK_MODIFICATION_TYPE_REPORT = "NetworkModification";
+
+    private final NetworkStoreService networkStoreService;
+
+    private final EquipmentInfosService equipmentInfosService;
+
+    private final ReportService reportService;
+
+    private final ApplicationContext context;
+
+    public NetworkModificationApplicator(NetworkStoreService networkStoreService, EquipmentInfosService equipmentInfosService,
+                                         ReportService reportService, ApplicationContext context) {
+        this.networkStoreService = networkStoreService;
+        this.equipmentInfosService = equipmentInfosService;
+        this.reportService = reportService;
+        this.context = context;
+    }
+
+    public List<ModificationInfos> applyModification(ModificationInfos modificationInfos, NetworkInfos networkInfos, ReportInfos reportInfos) {
+        NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkInfos.getNetworkUuuid(), networkStoreService, equipmentInfosService);
+        apply(modificationInfos, listener, reportInfos, ApplicationMode.UNITARY);
+        return listener.flushNetworkModifications();
+    }
+
+    public List<ModificationInfos> applyModifications(List<ModificationInfos> modificationInfosList, NetworkInfos networkInfos, ReportInfos reportInfos) {
+        NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkInfos.getNetworkUuuid(), networkStoreService, equipmentInfosService);
+        modificationInfosList.forEach(m -> apply(m, listener, reportInfos, ApplicationMode.MULTIPLE));
+        return listener.flushNetworkModifications();
+    }
+
+    public List<ModificationInfos> applyModifications(List<Pair<String, List<ModificationInfos>>> modificationInfosGroups, NetworkInfos networkInfos, UUID reportUuid) {
+        NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkInfos.getNetworkUuuid(), networkStoreService, equipmentInfosService);
+        modificationInfosGroups.forEach(g -> g.getRight().forEach(m -> apply(m, listener, new ReportInfos(reportUuid, g.getLeft()), ApplicationMode.MULTIPLE)));
+        return listener.flushNetworkModifications();
+    }
+
+    private void apply(ModificationInfos modificationInfos, NetworkStoreListener listener, ReportInfos reportInfos, ApplicationMode mode) {
+        String rootReporterId = reportInfos.getReporterId() + "@" + NETWORK_MODIFICATION_TYPE_REPORT;
+        ReporterModel reporter = new ReporterModel(rootReporterId, rootReporterId);
+        Reporter subReporter = modificationInfos.createSubReporter(reporter);
```

Comment:
moving any modification fails. Test Case:
- copy a load : modificationInfos is LoadCreationInfos => createSubReporter() works, apply works
- move a Load between 2 nodes (cut & paste): modificationInfos is base class ModificationInfos => createSubReporter() fails (exception TODO, apply is not performed)